### ISSUE TITLE
Finetuning possible?

### ISSUE THREAD

**Issue by turian** [facebookresearch/encodec#52]
## ❓ Questions

Is it possible to fine-tune the model? If so, how?

Or is only the compression/decompression model provided?

**Comment by jhauret** [comment]
I've fine-tuned Encodec's encoder and got interesting results for speech enhancement.

I simply fed my low quality signal to a fine-tuned Encodec's encoder while feeding a frozen Encodec's encoder with the reference audio. Then I just tried to minimize the L1 norm between the produced embeddings and keep using the same RVQ and decoder to produce enhanced audio from the acoustic tokens.

Of course, other methods are possible, and this can also be supplemented by using the streamable language model.

**Comment by listener17** [comment]
@jhauret interesting idea!

Do you happen to have the training code to train EnCodec?

**Comment by jhauret** [comment]
I have "a" training code that I do not want to share publicly for the moment (written with Pytorch Lightning, which is very handy to get started quickly).

**Comment by listener17** [comment]
@jhauret Understandable 😉 Thanks 🙏

**Comment by mikesol** [comment]
I hacked together a [jupyter notebook](https://colab.research.google.com/drive/1TXQ5O6dEG2b__F63ZwaBr5KQ_E8ybMs0?usp=sharing) that can give you a crude starting point (meaning it doesn't error out).
The loss function is complete rubbish though - their actual loss combines several, including https://github.com/facebookresearch/encodec/blob/0e2d0aed29362c8e8f52494baf3e6f99056b214f/encodec/msstftd.py (you can read more about it in the paper) using a novel balancing mechanism. 

There's also https://github.com/Mikxox/EnCodec_Trainer/blob/master/training.py, although I haven't personally tested it, but a lot of the important stuff seems to be there. It doesn't seem to use the balancer, though.

**Comment by DoodleHapes** [comment]
> I hacked together a [jupyter notebook](https://colab.research.google.com/drive/1TXQ5O6dEG2b__F63ZwaBr5KQ_E8ybMs0?usp=sharing) that can give you a crude starting point (meaning it doesn't error out). The loss function is complete rubbish though - their actual loss combines several, including https://github.com/facebookresearch/encodec/blob/0e2d0aed29362c8e8f52494baf3e6f99056b214f/encodec/msstftd.py (you can read more about it in the paper) using a novel balancing mechanism.
> 
> There's also https://github.com/Mikxox/EnCodec_Trainer/blob/master/training.py, although I haven't personally tested it, but a lot of the important stuff seems to be there. It doesn't seem to use the balancer, though.

Thanks for your commitment, I'm making a manage to finetune Encodec with your training code, and I encountered a trouble in the loss function which made my memory out of capacity. The memory keeps growing while the code running util trigger the out  of memory err. And I'm gonna to dive in the travel of code a new loss function to see if the memory question eliminated.
